python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1467,6 +1467,16 @@ def do_evaluate(
                 else:
                     (child,) = value.children
                 col = child.evaluate(df, context=ExecutionContext.GROUPBY).obj
+
+                if value.name in ["median", "mean", "quantile"] and col.type().id() in {
+                    plc.TypeId.DECIMAL128,
+                    plc.TypeId.DECIMAL64,
+                    plc.TypeId.DECIMAL32,
+                }:
+                    # libcudf doesn't support median (quantile) with decimal types,
+                    # but Polars returns a float result, so just cast the input.
+                    assert isinstance(child, expr.Col)
+                    col = plc.unary.cast(col, schema[child.name].plc)
```

Comment:
@Matt711 is this something that should happen in `decompose_single_agg`? I see it returns a "post_aggregate"` but this operation is like a "pre_aggregate"